auth/src/main/java/com/firebase/ui/auth/AuthUI.java

```diff
@@ -373,7 +380,84 @@ public Void then(@NonNull Task<GoogleApiClient> task) throws Exception {
         // Initialize SmartLock helper
         GoogleApiClientTaskHelper gacHelper = GoogleApiClientTaskHelper.getInstance(activity);
         gacHelper.getBuilder().addApi(Auth.CREDENTIALS_API);
-        CredentialsApiHelper credentialHelper = CredentialsApiHelper.getInstance(gacHelper);
+        CredentialApiHelper credentialHelper = CredentialsApiHelper.getInstance(gacHelper);
+
+        // Get all SmartLock credentials associated with the user
+        List<Credential> credentials = SmartLockBase.credentialsFromFirebaseUser(firebaseUser);
```

Comment:
From here down to the end of the function seems duplicated between the two `delete()` methods.  Can we abstract into a common private method?